tools/polarshift.rb

```diff
@@ -379,6 +395,93 @@ def setup_global_opts(options)
+    end
+
+    # the Custom fields object from the original run is the form of {key: value
+    # } Hash which is not suitable format for the query statement
+    # INPUT:
+    # [{"key"=>"assignee", "value"=>{"id"=>"pruan"}},
+    # {"key"=>"products", "value"=>{"id"=>"ocp"}},
+    # {"key"=>"version", "value"=>{"id"=>"4_3"}},
+    # {"key"=>"plannedin", "value"=>{"id"=>"OCP_4_3_Feature_Feeze"}},
+    # {"key"=>"caseimportance", "value"=>{"EnumOptionId"=>[{"id"=>"critical"}, {"id"=>"high"}, {"id"=>"medium"}, {"id"=>"low"}]}},
+    # {"key"=>"tags", "value"=>"ocp43ff13"},
+    # {"key"=>"env_container_runtime", "value"=>{"id"=>"crio_1x"}},
+    # {"key"=>"env_install_method", "value"=>{"id"=>"ipi"}},
+    # {"key"=>"env_iaas_cloud_provider", "value"=>{"id"=>"azure"}},
+    # {"key"=>"env_registry_storage_type", "value"=>{"id"=>"s3"}},
+    # {"key"=>"subteam", "value"=>{"EnumOptionId"=>{"id"=>"metering"}}},
+    # {"key"=>"caseautomation", "value"=>{"EnumOptionId"=>[{"id"=>"manualonly"}, {"id"=>"notautomated"}]}},
+    # {"key"=>"env_network_backend", "value"=>{"id"=>"openshift-sdn"}},
+    # {"key"=>"env_cluster", "value"=>{"EnumOptionId"=>{"id"=>"ocp_cluster"}}}]
+    #
+    # OUTPUT:
+    # {"assignee"=>"pruan",
+    #  "products"=>"ocp",
+    #  "version"=>"4_3",
+    #  "plannedin"=>"OCP_4_3_Feature_Feeze",
+    #  "caseimportance"=>["critical", "high", "medium", "low"],
+    #  "tags"=>"ocp43ff13",
+    #  "env_container_runtime"=>"crio_1x",
+    #  "env_install_method"=>"ipi",
+    #  "env_iaas_cloud_provider"=>"azure",
+    #  "env_registry_storage_type"=>"s3",
+    #  "subteam"=>"metering",
+    #  "caseautomation"=>["manualonly", "notautomated"],
+    #  "env_network_backend"=>"openshift-sdn",
+    #  "env_cluster"=>"ocp_cluster"}
+    # given a query_result Custom object (Array).
+    # @return Hash to be constructed
+    def transform_custom_fields(src)
+      custom_fields = {}
+      src.each do |row_data|
+        if row_data['value'].is_a? Hash
+          if row_data['value'].has_key? 'EnumOptionId'
+            if row_data['value']['EnumOptionId'].is_a? Hash
+              value = row_data['value']['EnumOptionId']['id']
+            else
+              value = row_data['value']['EnumOptionId'].map {|r| r['id'] }
+            end
+          else
+            value = row_data['value']['id']
+          end
+        else
+          value = row_data['value']
+        end
+        # in case of nil, just force it to be empty string
+        value ||= ""
+        custom_fields[row_data['key']] = value
+      end
+      return custom_fields
+    end
+
+    # given a query_result
+    # @return a YAML string that will be feed into create_run to clone a past
+    #         polarion run
+    # The YAML basically contains 3 fields
+    # 1.   "run_title":  "Clone of xyz run"
+    # 2.   "case_query": "<SQL_STATEMENT>"
+    # 3.   "custom_fields": <CONSTRUCTED_FROM_call_to_get_smart_run_method>
```

Comment:
I think info above is outdated. Please update it.

Other than that looks nice!

Reply:
@akostadinov comment updated,  PTAL again.